src/layer/DivOverlay.js

```diff
@@ -230,23 +230,20 @@ export var DivOverlay = Layer.extend({
 	},
 
 	// prepare bound overlay to open: update latlng pos / content source (for FeatureGroup)
-	_prepareOpen: function (latlng) {
-		var source = this._source;
+	_prepareOpen: function (latlng, parent) {
+		var source = parent;
 		if (!source._map) { return false; }
 
 		if (source instanceof FeatureGroup) {
 			source = null;
-			var layers = this._source._layers;
+			var layers = parent._layers;
 			for (var id in layers) {
-				if (layers[id]._map) {
+				if (layers[id]._map && (Number(id) === this._source._leaflet_id || this._source instanceof FeatureGroup)) {

```

Comment:
I'd rather use the `stamp` function from `Util`, instead of querying `_leaflet_id`. It's what the codebase currently does. 

The only reason not to do so would be a use case where `this._source` is not a `L.Layer` - I don't think that's possible here.

Reply:
@Falke-Design That was both to select the correct layer as well as to preserve the behavior of FeatureGroups. However, your idea was a better way to select the correct layer; I just had to add a little bit to it to preserve FeatureGroup behavior.